src/sentry/static/sentry/app/views/organizationSettings.jsx

```diff
@@ -0,0 +1,356 @@
+import IndicatorStore from '../stores/indicatorStore';
+import LoadingIndicator from '../components/loadingIndicator';
+import OrganizationHomeContainer from '../components/organizations/homeContainer';
+import {t} from '../locale';
+
+
+const OrganizationSettingsForm = React.createClass({
+  propTypes: {
+    orgId: React.PropTypes.string.isRequired,
+    access: React.PropTypes.object.isRequired,
+    initialData: React.PropTypes.object.isRequired,
+    onSave: React.PropTypes.func.isRequired
+  },
+
+  mixins: [ApiMixin],
+
+  getInitialState() {
+    return {
+      formData: this.buildFormData(this.props.initialData),
+      errors: {},
+      hasChanges: false,
+    };
+  },
+
+  buildFormData(data) {
+    let result = {
+      name: data.name,
+      slug: data.slug,
+      openMembership: data.openMembership,
+      allowSharedIssues: data.allowSharedIssues,
+      isEarlyAdopter: data.isEarlyAdopter,
+      enhancedPrivacy: data.enhancedPrivacy,
+      dataScrubber: data.dataScrubber,
+      dataScrubberDefaults: data.dataScrubberDefaults,
+      scrubIPAddresses: data.scrubIPAddresses,
+      safeFields: data.safeFields.join('\n'),
+      sensitiveFields: data.sensitiveFields.join('\n'),
+    };
+    if (this.props.access.has('org:delete')) {
+      result.defaultRole = data.defaultRole;
+    }
+    return result;
+  },
+
+  onFieldChange(name, value) {
+    let formData = this.state.formData;
+    formData[name] = value;
+    this.setState({
+      hasChanges: true,
+      formData: formData,
+    });
+  },
+
+  onSubmit(e) {
+    e.preventDefault();
+
+    if (this.state.state == FormState.SAVING) {
+      return;
+    }
+
+    this.setState({
+      state: FormState.SAVING,
+      hasChanges: false,
+    }, () => {
+      let loadingIndicator = IndicatorStore.add(t('Saving changes..'));
+      let {orgId} = this.props;
+      let formData = this.state.formData;
+      this.api.request(`/organizations/${orgId}/`, {
+        method: 'PUT',
+        data: {
+          ...formData,
+          safeFields: formData.safeFields.split('\n'),
+          sensitiveFields: formData.sensitiveFields.split('\n'),
+        },
+        success: (data) => {
+          this.props.onSave(data);
+          this.setState({
+            state: FormState.READY,
+            errors: {},
+          });
+          IndicatorStore.remove(loadingIndicator);
+          IndicatorStore.add(t('Changes saved.'), 'success', {
+            duration: 1500
+          });
+        },
+        error: (error) => {
+          this.setState({
+            state: FormState.ERROR,
+            errors: error.responseJSON,
+          });
+          IndicatorStore.remove(loadingIndicator);
+          IndicatorStore.add(t('Unable to save changes. Please try again.'), 'error', {
+            duration: 3000
+          });
+        },
+      });
+    });
+  },
+
+  render() {
+    let isSaving = this.state.state === FormState.SAVING;
+    let errors = this.state.errors;
```

Comment:
Totally stylistic, but I'd probably do:

```javascript
let {errors, formData} = this.state;
let {access, initialData} = this.props;
```

At a glance it's nice to see what's coming from props, what's coming from state. But this is purely stylistic / can be ignored entirely.